### DISCUSSION TITLE
Kornia warp_frame_depth confusion

### DISCUSSION THREAD

**Discussion by akristoffersen** [kornia/kornia#1672] [Questions]
Hello! I have a quick question about [warp_frame_depth](https://kornia.readthedocs.io/en/latest/geometry.depth.html#kornia.geometry.depth.warp_frame_depth). It says in the spec that the depth argument is in the destination frame, but I'm not completely sure what that means--

Say you had a right image R and a left image L, and you wanted to recreate L by warping R. Do you input a depth map of image L or  of R?

Looking around at its use in the wild, it seems like people input the depth map associated with the input image, but from the spec it seems like you'd want the depth map of the destination? I can't really think of a situation where you'd want to reproject an image to a new view where you have a depth map but no actual image.

**Comment by edgarriba** [comment]
that function was originally designed to solve the specific problem to optimize the depth via gradient descent as showed here: https://github.com/kornia/kornia-examples/blob/master/depth_estimation.ipynb

In that case we assumed to have a fixed depth on a reference frame, and set of arbitrary image (or features) with their respective calibration. We can adjust any time the convention with a proper use case and interface design. 

I'm revisiting the part ot 3d geometry this days to include liealgebra and more camera models, so happy to spend some time on this topic.

**Comment by akristoffersen** [comment]
That would be great! It would be incredibly useful to have some sort of forward reprojection from depth method, as it looks at least some people are using this method incorrectly as such ([example](https://github.com/tpunnoose/monocular-depth-estimation/blob/d148d403d959a479975ed16f688cb71750b84b44/loss.py)).

**Reply by edgarriba** [reply]
oh, I see. Will have to create maybe new examples about this topic
